src/Management/src/Endpoint/ConfigureActuatorsMiddlewareStartupFilter.cs

```diff
@@ -33,9 +45,30 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
                 app.UseCloudFoundrySecurity();
             }
 
+            int? beforeMiddlewareCount = GetMiddlewareCount(app);
             next.Invoke(app);
+            int? afterMiddlewareCount = GetMiddlewareCount(app);
+
+            if (beforeMiddlewareCount != afterMiddlewareCount)
+            {
+                _logger.LogWarning(
+                    "Your app adds custom middleware to the pipeline, while actuators were registered to auto-perform middleware setup. This combination is usually undesired. " +
+                    "To correct this, either remove your middleware setup code, or register actuators by setting their configureMiddleware parameter to false.");
+            }
 
             app.UseActuatorEndpoints();
         };
     }
+
+    private static int? GetMiddlewareCount(IApplicationBuilder app)
+    {
+        FieldInfo? componentsField = app.GetType().GetField("_components", BindingFlags.NonPublic | BindingFlags.Instance);
+        return componentsField?.GetValue(app) is List<Func<RequestDelegate, RequestDelegate>> components ? components.Count(IsMiddleware) : null;
+    }
+
+    private static bool IsMiddleware(Func<RequestDelegate, RequestDelegate> component)
+    {
+        // This type exists so that ASP.NET Core can identify where to inject UseRouting. It is not a real middleware.
+        return component.Target == null || component.Target.ToString() != "Microsoft.AspNetCore.Builder.WebApplicationBuilder+WireSourcePipeline";
```

Comment:
This seems like it casts a fairly wide net, can/should we only look for `UsePathBaseMiddleware`?